redaxo/src/addons/be_style/plugins/redaxo/assets/javascripts/redaxo.js

```diff
@@ -161,6 +161,42 @@
         };
     })();
 
+    // -----------------------------------------------------------------------
+
+    // Darkmode
+    // prerequisites to detect darkmode-changes
+    if( rex && rex.darkmode && !redaxo.darkmode && window.matchMedia ) {
+
+        redaxo.darkmode = {
+            match: window.matchMedia('(prefers-color-scheme: dark)'),
+            mode: null,
+            switched: function( mode ) {
+                let darkmode;
+                if( document.body.classList.contains('rex-theme-light') ){
+                    darkmode = false;
+                } else if ( document.body.classList.contains('rex-theme-dark') ){
+                    darkmode = true;
+                } else {
+                    darkmode = true === mode;
```

Comment:
sollte der `mode` parameter, wenn gegeben, der body-class überwiegen?

oder ist diese methode nicht als "setter" gedacht?

Reply:
Wenn ich die Logik des Darkmode in REDAXO richtig verstehe, wird der Mode ("dark/light") entweder per body-Attribut class vorgegeben oder wenn nicht vorgegeben ("auto") durch das System bestimmt. Daher auch hier die Reihenfolge, erst die Klassen abzufragen. Das erwischt "dark/light" als Festeinstellung. Ist keine der beiden Klassen gesetzt, ist also "auto" die Vorgabe und dann wird eben der mode-Wert gesetzt.